s3tests_boto3/functional/test_s3.py

```diff
@@ -7509,6 +7509,40 @@ def test_versioning_obj_suspend_versions():
     assert len(version_ids) == 0
     assert len(version_ids) == len(contents)
 
+def test_versioning_obj_suspended_copy():
+    bucket_name = get_new_bucket()
+    client = get_client()
+
+    check_configure_versioning_retry(bucket_name, "Enabled", "Enabled")
+
+    key1 = 'testobj1'
+    num_versions = 1
+    (version_ids, contents) = create_multiple_versions(client, bucket_name, key1, num_versions)
+
+    check_configure_versioning_retry(bucket_name, "Suspended", "Suspended")
+
+    content = 'null content'
+    overwrite_suspended_versioning_obj(client, bucket_name, key1, version_ids, contents, content)
+
+    # copy to another object
+    key2 = 'testobj2'
+    copy_source = {'Bucket': bucket_name, 'Key': key1}
+    client.copy_object(Bucket=bucket_name, Key=key2, CopySource=copy_source)
+
+    # delete the source object. keep the 'null' entry in version_ids
+    client.delete_object(Bucket=bucket_name, Key=key1)
+
+    # get the target object
+    response = client.get_object(Bucket=bucket_name, Key=key2)
```

Comment:
when [this job](http://qa-proxy.ceph.com/teuthology/cbodley-2024-02-11_16:11:11-rgw-wip-cbodley2-testing-distro-default-smithi/7556252/teuthology.log) ran against dbstore in the rgw/dbstore suite, it failed here with:
> An error occurred (NoSuchKey) when calling the GetObject operation: None

not sure what exactly dbstore is doing wrong here, but lets tag this test with `@pytest.mark.fails_on_dbstore` for now

Reply:
done